binary-decode.go

```diff
@@ -695,18 +696,158 @@ func (cdc *Codec) decodeReflectBinaryStruct(bz []byte, info *TypeInfo, rv reflec
 
 		// Read "StructTerm".
 		// NOTE: In the future, we'll need to break out of a loop
-		// when encoutering an StructTerm typ3 byte.
-		var typ = Typ3(0x00)
+		// when encountering an StructTerm typ3 byte.
+
+		var typ Typ3
+		// we have read all, or do we? consume the rest
+		for len(bz) > 2 && bz[0]&0xF8 != 0 && Typ3(bz[0]) != Typ3_StructTerm {
+			fnum, typ3, _n, _ := decodeFieldNumberAndTyp3(bz)
+			if slide(&bz, &n, _n) && err != nil {
```

Comment:
This err is the wrong err to check, as it wasn't assigned from `decodeFieldNumberAndTyp3`.

Reply:
fixed